### ISSUE TITLE
[Bug]: Change max value of dimension, insert() succeed but create_index() failed

### ISSUE THREAD

**Issue by yhmo** [milvus-io/milvus#30427]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.3.7
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

There is a parameter checker for create_index(): https://github.com/milvus-io/milvus/blob/77e123762fff791c2d86f21a61413c9089045b61/pkg/util/indexparamcheck/base_checker.go#L12
The checker uses a hard-coded value to verify dimension value. The "DefaultMaxDim" is hard-coded to 32768: https://github.com/milvus-io/milvus/blob/77e123762fff791c2d86f21a61413c9089045b61/pkg/util/indexparamcheck/constraints.go#L21

The proxy is using the proxy.maxDimension to verify dimension of inserted vectors. So, insert() request succeeds after you enlarge the proxy.maxDimension. But create_index() failed due to this hard code value.

### Expected Behavior

Knowhere doesn't limit dimension, the max dimension should be configured-able.
After enlarging the proxy.maxDimension in milvus.yaml, the server should allow large dimension vectors to be inserted and indexed.

### Steps To Reproduce

```markdown
1. change proxy.maxDimension to 100000 in milvus.yaml
2. start the server
3. insert vectors with dimension=100000
4. create index
```

**Comment by yanliang567** [comment]
@liliu-z does knowhere has limitation in dimension for different index types? is it safe to update the max dim beyond 32768?

**Comment by yanliang567** [comment]
keep it until verification done

**Comment by yanliang567** [comment]
@elstic I believe this was fixed on milvus 2.4, please help to confirm

**Comment by elstic** [comment]
@yhmo This issue has been fixed in the 2.4 branch.
